### DISCUSSION TITLE
Move custom preference out of the "Extensions" folder

### DISCUSSION THREAD

**Discussion by Hanksha** [eclipse-theia/theia#14353] [Q&A] [answered]
Hi,

I'd like to know if there is a way to move custom preferences (from my own theia extension) out of the "Extensions" folder in the settings editor?
![image](https://github.com/user-attachments/assets/6336cb66-1e92-4d00-b878-dcdade326bb9)

**Comment by msujew** [comment, marked as answer]
Hey @Hanksha,

you will need to override the `PreferenceLayoutProvider` and return a modified layout, see here:

https://github.com/eclipse-theia/theia/blob/c1e4279cf85aff0b4fdfccbe605fc391aa599e0a/packages/preferences/src/browser/util/preference-layout.ts#L326-L328

**Comment by Hanksha** [comment]
Hi @msujew 

Thanks that worked, but I could not use a "title" with it to group the properties together (in the schema), is there a way to do that?

**Reply by msujew** [reply]
I cannot really follow. What do you expect the result to look like?

**Comment by Hanksha** [comment]
I'd like them grouped like this:
![image](https://github.com/user-attachments/assets/67318475-5be3-4585-9a19-00a707c486e5)

But I get:
![image](https://github.com/user-attachments/assets/7710968c-e5b7-4ac0-aaa0-2f54a221f4b3)

Layout:

```
const MARTINI_LAYOUT: PreferenceLayout[] = [
    {
        id: "martini",
        label: "Martini Designer",
        settings: ["martini.*"]
    },
    ...DEFAULT_LAYOUT
];
```

Schema:
```
export const rumPreferenceSchema: PreferenceSchema = {
    type: "object",
    properties: {
        "martini.sendDiagnosticData": {
            type: "boolean",
            default: true,
            description: ""
        },
        "martini.includeUserData": {
            type: "boolean",
            default: true,
            description: ""
        }
    }
};
```

If I set the title in the schema, they are grouped under Data Collection but they end up in "Extensions" folder and if I remove the title, they are all under "Martini Designer" but I'd like to have "Martini Designer" > "Data Collection" as a structure.

**Reply by Hanksha** [reply]
Ah I think I got it @msujew, I need do define the layout like this?

```
const MARTINI_LAYOUT: PreferenceLayout[] = [
    {
        id: "martini",
        label: "Martini Designer",
        settings: ["martini.*"],
        children: [
            {
                id: "martini.dataCollection",
                label: "Data Collection",
                settings: ["martini.dataCollection.*"]
            }
        ]
    },
    ...DEFAULT_LAYOUT
];
```

The only downside is that I have to define all statically in one spot while I broke down most features into multiple extensions. I guess I can just change the behavior of PreferenceLayoutProvider and introduce a contribution API for the layout instead.
